apps/aehttp/src/aehttp_dispatch_rosetta.erl

```diff
@@ -0,0 +1,659 @@
+        {200, [], Resp}
+    catch
+        block_not_found ->
+            {200, [], rosetta_error_response(?ROSETTA_ERR_BLOCK_NOT_FOUND)};
+        tx_not_found ->
+            {200, [], rosetta_error_response(?ROSETTA_ERR_TX_NOT_FOUND)}
+    end;
+handle_request_(call, _, _Context) ->
+    {501, [], #{}};
+handle_request_(constructionCombine, _, _Context) ->
+    {501, [], #{}};
+handle_request_(constructionDerive,
+                #{'ConstructionDeriveRequest' :=
+                    #{<<"network_identifier">> := #{<<"blockchain">> := <<"aeternity">>},
+                     <<"public_key">> := #{<<"curve_type">> := CurveType,
+                                           <<"hex_bytes">> := HexBytes}}},
+                _Context) ->
+    case CurveType of
+        <<"edwards25519">> ->
+            PKBytes = aeu_hex:hex_to_bin(HexBytes),
+            Account = aeapi:format(account_pubkey, PKBytes),
+            AccountIdentifier = #{<<"account_identifier">> =>
+                                    #{<<"address">> => Account}},
+            Resp = #{<<"ConstructionDeriveResponse">> => AccountIdentifier},
+            {200, [], Resp};
+        _ ->
+            {501, [], #{}}
+    end;
+handle_request_(constructionHash, _, _Context) ->
+    {501, [], #{}};
+handle_request_(constructionMetadata, _, _Context) ->
+    {501, [], #{}};
+handle_request_(constructionParse, _, _Context) ->
+    {501, [], #{}};
+handle_request_(constructionPayloads, _, _Context) ->
+    {501, [], #{}};
+handle_request_(constructionPreprocess, _, _Context) ->
+    {501, [], #{}};
+handle_request_(constructionSubmit, _, _Context) ->
+    {501, [], #{}};
+handle_request_(eventsBlocks, _, _Context) ->
+    {501, [], #{}};
+handle_request_(mempool,
+                #{'NetworkRequest' :=
+                      #{<<"network_identifier">> := #{<<"blockchain">> := <<"aeternity">>}}},
+                _Context) ->
+    {ok, SignedTxList} = aec_tx_pool:peek(infinity),
+    SignedTxHashList =
+        [#{<<"hash">> => aeapi:format(tx_hash, aetx_sign:hash(X))} || X <- SignedTxList],
+    Resp = #{<<"transaction_identifiers">> => SignedTxHashList},
+    {200, [], Resp};
+handle_request_(mempoolTransaction,
+                #{'MempoolTransactionRequest' :=
+                      #{<<"network_identifier">> := #{<<"blockchain">> := <<"aeternity">>},
+                        <<"transaction_identifier">> := #{<<"hash">> := TxHash}}},
+                _Context) ->
+    %%
+    %% Decode the transaction hash
+    %%
+    TxHashInternal =
+        case aeapi:decode(TxHash) of
+            {tx_hash, TxHash0} ->
+                TxHash0;
+            _ ->
+                throw(tx_not_found)
+        end,
+    case aec_db:find_tx_location(TxHashInternal) of
+        mempool ->
+            ok;
+        _ ->
+            throw(tx_not_found)
+    end,
+    case aec_db:find_signed_tx(TxHashInternal) of
+        {value, SignedTx} ->
```

Comment:
Should be done within one activity

Reply:
As discussed will leave this as is for the moment. It's not performance critical and is clear enough